skimage/filters/rank/core_cy.pyx

```diff
@@ -271,3 +286,293 @@ cdef void _core(void kernel(dtype_t_out*, Py_ssize_t, Py_ssize_t*, double,
+        se_e_c = <Py_ssize_t*>malloc(se_size)
+        se_e_p = <Py_ssize_t*>malloc(se_size)
+
+        se_w_r = <Py_ssize_t*>malloc(se_size)
+        se_w_c = <Py_ssize_t*>malloc(se_size)
+        se_w_p = <Py_ssize_t*>malloc(se_size)
+
+        se_n_r = <Py_ssize_t*>malloc(se_size)
+        se_n_c = <Py_ssize_t*>malloc(se_size)
+        se_n_p = <Py_ssize_t*>malloc(se_size)
+
+        se_s_r = <Py_ssize_t*>malloc(se_size)
+        se_s_c = <Py_ssize_t*>malloc(se_size)
+        se_s_p = <Py_ssize_t*>malloc(se_size)
+
+        histo = <Py_ssize_t*>malloc(n_bins * sizeof(Py_ssize_t))
+
+        if (se_e_r is NULL or se_e_c is NULL or se_e_p is NULL or
+                se_w_r is NULL or se_w_c is NULL or se_w_p is NULL or
+                se_n_r is NULL or se_n_c is NULL or se_n_p is NULL or
+                se_s_r is NULL or se_s_c is NULL or se_s_p is NULL or
+                histo is NULL):
+            free(se_e_r)
+            free(se_e_c)
+            free(se_e_p)
+            free(se_w_r)
+            free(se_w_c)
+            free(se_w_p)
+            free(se_n_r)
+            free(se_n_c)
+            free(se_n_p)
+            free(se_s_r)
+            free(se_s_c)
+            free(se_s_p)
+            free(histo)
+            with gil:
+                raise MemoryError()
+
+        for i in range(n_bins):
+            histo[i] = 0
+
+        num_se_n = num_se_s = num_se_e = num_se_w = 0
+
+        for r in range(srows):
+            for c in range(scols):
+                for p in range(splanes):
+                    if t_e[p, r, c]:
+                        se_e_r[num_se_e] = r - centre_r
+                        se_e_c[num_se_e] = c - centre_c
+                        se_e_p[num_se_e] = p - centre_p
+                        num_se_e += 1
+                    if t_w[p, r, c]:
+                        se_w_r[num_se_w] = r - centre_r
+                        se_w_c[num_se_w] = c - centre_c
+                        se_w_p[num_se_w] = p - centre_p
+                        num_se_w += 1
+                    if t_n[p, r, c]:
+                        se_n_r[num_se_n] = r - centre_r
+                        se_n_c[num_se_n] = c - centre_c
+                        se_n_p[num_se_n] = p - centre_p
+                        num_se_n += 1
+                    if t_s[p, r, c]:
+                        se_s_r[num_se_s] = r - centre_r
+                        se_s_c[num_se_s] = c - centre_c
+                        se_s_p[num_se_s] = p - centre_p
+                        num_se_s += 1
+
+        for p in range(planes):
+            for i in range(n_bins):
+                histo[i] = 0
+            pop = 0
+
+            for r in range(srows):
+                for c in range(scols):
+                    for j in range(splanes):
+                        rr = r - centre_r
+                        cc = c - centre_c
+                        pp = j - centre_p + p
+
+                        # change this to accept 3D
```

Comment:
What does this comment mean? Is it outdated?

Reply:
I believe so. @ryanlu41 ?

Reply:
yes i believe so, apologies

Reply:
fixed